### PR TITLE
Issue #6264: Make mobile menu toggle text configurable.

### LINKED ISSUE #6264: Make mobile menu toggle text configurable
## Description of the need

Right now the mobile menu toggle text is hardcoded (though translated) as `Menu`. Does it seem reasonable to folks to allow that text to be adjusted on a per menu block basis? (ie. when configuring the block, it defaults to `Menu` but can be overwritten)
```
      $data['content']['#prefix'] = theme('menu_toggle', array(
        'enabled' => $config['toggle'],
        'id' => $id,
        'text' => t('Menu'),
      ));
```

I am considering a case where there are potentially two toggles (in different parts of the page) and it would be helpful to be able to use different language for each. At first I thought I'd just override `theme_menu_toggle` but I really don't have enough information to grab onto inside that function to know much about the particulars of the menu.

If we add this, it could only appear underneath the "Display menu toggle..." option if it was selected:

image

### PR DESCRIPTION
**Pull request by laryn**
Fixes https://github.com/backdrop/backdrop-issues/issues/6264

### CODE REVIEW

File: core/modules/system/system.menu.inc [hunk 1]
```diff
@@ -27,6 +27,7 @@ function system_menu_block_defaults($menu_name) {
     'expand_all' => FALSE,
     'clickdown' => FALSE,
     'collapse' => 'default',
+    'toggle_text' => t('Menu'),
```
Comment (herbdool) [inline core/modules/system/system.menu.inc]: I don't think we want to store the translated string here. It needs to be just 'Menu' so it gets checked in the theme function where you also have it wrapped in `t()`.

**Review by klonos — APPROVED** [review]

File: core/modules/system/system.menu.inc (hunk 1 above)
Comment (laryn) [reply core/modules/system/system.menu.inc]: Good catch.

File: core/includes/menu.inc [hunk 2]
```diff
@@ -1905,11 +1905,12 @@ function theme_menu_local_actions($variables) {
 function theme_menu_toggle($variables) {
   $output = '';
   if ($variables['enabled']) {
+    $toggle_text = t(check_plain($variables['text']));
```
Comment (klonos) [inline core/includes/menu.inc]: Since this is a bit of an odd use of `t()`, can we please add an inline comment here to explain why we are doing this?

File: core/modules/system/system.menu.inc [hunk 3]
```diff
@@ -146,6 +147,17 @@ function system_menu_block_form($config) {
       '!menu_settings' => $link,
     )),
   );
+  $form['toggle_text'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Toggle button text'),
+    '#default_value' => $config['toggle_text'],
+    '#description' => t('Text to display beside the toggle button. Default: <em>Menu</em>.'),
```
Comment (klonos) [inline core/modules/system/system.menu.inc]: Can we please make this a placeholder that is pulling its value from `system_menu_block_defaults()`?

File: core/modules/system/system.menu.inc [hunk 4]
```diff
@@ -146,6 +147,17 @@ function system_menu_block_form($config) {
...
+    '#states' => array(
```
Comment (klonos) [inline core/modules/system/system.menu.inc:159]: Lets also add the relatively new `#indentation` property here, to nest the textfield under the "Display menu toggle button on small screens" checkbox:
```suggestion
    '#indentation' => 1,
    '#states' => array(
```

File: core/modules/system/system.menu.inc (hunk 3 above)
Comment (laryn) [reply core/modules/system/system.menu.inc]: @klonos 
Can you clarify? The toggle text default is already in `system_menu_block_defaults()`:

```
'toggle_text' => 'Menu',
```

File: core/includes/menu.inc (hunk 2 above)
Comment (laryn) [reply core/includes/menu.inc]: @klonos Sure, I've added one.

File: core/modules/system/system.menu.inc (hunk 3 above)
Comment (bugfolder) [reply core/modules/system/system.menu.inc]: @laryn, I think @klonos means something like this:

```suggestion
    '#description' => t('Text to display beside the toggle button. Default: %toggle_text.', array('%toggle_text' => system_menu_block_defaults($config['menu_name'])['toggle_text'])),
```

I've squeezed it into a single line of code for suggestion purposes, but it might be more readable to create a `$toggle_text` variable just before the form entry, then plug it into the `#description` `t()` call. And note that since we're using `t()`, we can use a `%` variable to get the emphasis, so we don't need to add HTML inside the translatable text.

Comment (laryn) [reply core/modules/system/system.menu.inc]: Thanks @bugfolder -- resolved in the latest commit on this PR.

File: core/includes/menu.inc [hunk 5]
```diff
@@ -1905,11 +1905,13 @@ function theme_menu_local_actions($variables) {
 function theme_menu_toggle($variables) {
   $output = '';
   if ($variables['enabled']) {
+    // We are allowing the user-entered menu toggle text to be translated.
```
Comment (klonos) [inline core/includes/menu.inc]: Lets shorten this to:
```suggestion
    // Allow the user-entered menu toggle text to be translated.
```

File: core/includes/menu.inc (hunk 2 above)
Comment (quicksketch) [reply core/includes/menu.inc]: I think this should be reversed so that translators don't get HTML-encoded strings in their translations.

```suggestion
    $toggle_text = check_plain(t($variables['text']));
```

Comment (quicksketch) [reply core/includes/menu.inc:1917]: Actually this should probably be outside the theme function and be moved up to `system_menu_block_build()`.

**Comment by quicksketch** [comment]
Merged into 1.x for 1.27.0. Thanks @laryn!
